libs/frontend/ui-build/src/components/BuildDndOverlay.vue

```diff
@@ -79,35 +78,163 @@ const overlayStyle = computed(() => {
-          overlayTop = elementRelativeBottom - outlineThickness
+          overlayTop = relativeBottom - outlineThickness
           overlayWidth = elementWidth
           overlayHeight = outlineThickness
           overlayBackgroundColor = outlineColor
         }
       }
 
-      // Divide offsets by `builderScale` because `.build-content-window-inner` scales
-      // based on `builderScale` in the editor context.
-      overlayTop /= builderScale
-      overlayLeft /= builderScale
-      overlayWidth /= builderScale
-      overlayHeight /= builderScale
-
-      overlayTop += buildContentWindowInner.scrollTop
-      overlayLeft += buildContentWindowInner.scrollLeft
+      const { top, left, width, height } = normalizeDimensions(
+        overlayTop,
+        overlayLeft,
+        overlayWidth,
+        overlayHeight,
+        buildContentWindowInner,
+      )
 
       return {
-        top: `${overlayTop}px`,
-        left: `${overlayLeft}px`,
-        width: `${overlayWidth}px`,
-        height: `${overlayHeight}px`,
+        top: `${top}px`,
+        left: `${left}px`,
+        width: `${width}px`,
+        height: `${height}px`,
         backgroundColor: overlayBackgroundColor,
         border: overlayBorder,
       }
     }
   }
   return undefined
 })
+
+// Use `ref`+`watch` instead of `computed` because we have to wait for the screen to (re)render
+// before calculating the width and height of the component on page load & after selection.
+const selectionOverlayStyle = ref()
+
+const computeSelectionOverlay = async (selectedComponent: IComponent | undefined) => {
+  if (selectedComponent) {
+    // This sleep is necessary for buildContentWindowInner and componentElement
+    await sleep(1)
+
+    const buildContentWindowInner = document.getElementById(buildContentWindowInnerId)
+    let componentElement = document.getElementById(selectedComponent.id)
+    // Container size might be different from the actual component
+    const containerChild = componentElement?.children?.[0]
+    if (containerChild?.className?.includes('component-content-container')) {
+      componentElement = containerChild.children?.[0] as HTMLElement
+    }
+
+    if (buildContentWindowInner && componentElement) {
+      const { top: buildContentWindowTop, left: buildContentWindowLeft } =
+        buildContentWindowInner.getBoundingClientRect()
+
+      const {
+        top: elementTop,
+        left: elementLeft,
+        width: elementWidth,
+        height: elementHeight,
+      } = componentElement.getBoundingClientRect()
+
+      const elementRelativeTop = elementTop - buildContentWindowTop
+      const elementRelativeLeft = elementLeft - buildContentWindowLeft
+
+      const { top, left, width, height } = normalizeDimensions(
+        elementRelativeTop,
+        elementRelativeLeft,
+        elementWidth,
+        elementHeight,
+        buildContentWindowInner,
+      )
+
+      selectionOverlayStyle.value = {
+        top: `${top - 1}px`,
```

Comment:
This `-1` appears to position the border correctly in all cases, but I don't know why it's necessary. I also increased the border width to 2px.